master/internal/webhooks/postgres_webhook.go

```diff
@@ -272,6 +280,121 @@ func (l *WebhookManager) deleteWebhook(ctx context.Context, id WebhookID) error
+	if err != nil {
+		return fmt.Errorf("error getting experiment from id %d: %w", experimentID, err)
+	}
+
+	activeConfig, err := expconf.ParseAnyExperimentConfigYAML(m.ConfigBytes)
+	if err != nil {
+		return fmt.Errorf("error parsing experiment config: %w", err)
+	}
+	integrationsConfig := activeConfig.Integrations()
+	if integrationsConfig == nil {
+		return nil
+	}
+	webhookConfig := integrationsConfig.Webhooks
+	if webhookConfig == nil || webhookConfig.Exclude {
+		return nil
+	}
+
+	workspaceID, err := experiment.GetWorkspaceFromExperiment(ctx, &m.Experiment)
+	if err != nil {
+		return fmt.Errorf("error getting workspace from experiment : %w", err)
+	}
+
+	var es []Event
+	if webhookConfig.WebhookID != nil {
+		for _, webhookID := range *webhookConfig.WebhookID {
+			webhook, err := GetWebhook(ctx, webhookID)
+			if err != nil {
+				return fmt.Errorf("error getting webhook from id %d: %w", webhookID, err)
+			}
+			err = generateEventForCustomTrigger(
+				ctx, &es, webhook.Triggers, webhook.WebhookType, webhook.URL, m.Experiment, activeConfig, data, trialID)
+			if err != nil {
+				return fmt.Errorf("error genrating event for webhook with ID %d %+v: %w", webhookID, webhook, err)
+			}
+		}
+	}
+	if webhookConfig.WebhookName != nil {
+		for _, webhookName := range *webhookConfig.WebhookName {
+			webhook, err := getWebhookByName(ctx, webhookName, int(workspaceID))
+			if err != nil {
+				return fmt.Errorf("error getting webhook from name %s: %w", webhookName, err)
+			}
+			if webhook == nil {
+				continue
+			}
+			err = generateEventForCustomTrigger(
+				ctx, &es, webhook.Triggers, webhook.WebhookType, webhook.URL, m.Experiment, activeConfig, data, trialID)
+			if err != nil {
+				return fmt.Errorf("error genrating event %s %+v: %w", webhookName, webhook, err)
+			}
+		}
+	}
+
+	if len(es) == 0 {
+		return nil
+	}
+
+	if _, err := db.Bun().NewInsert().Model(&es).Exec(ctx); err != nil {
+		return fmt.Errorf("report experiment state changed inserting event trigger: %w", err)
+	}
+
+	singletonShipper.Wake()
+	return nil
+}
+
+func generateEventForCustomTrigger(
+	ctx context.Context,
+	es *[]Event,
+	triggers Triggers,
+	webhookType WebhookType,
+	webhookURL string,
+	e model.Experiment,
+	activeConfig expconf.ExperimentConfig,
+	data CustomTriggerData,
+	trialID *int,
+) error {
+	for _, t := range triggers {
+		if t.TriggerType != TriggerTypeCustom {
+			continue
+		}
+		p, err := generateEventPayload(
+			ctx, webhookType, e, activeConfig, e.State, TriggerTypeCustom, &data, trialID,
+		)
+		if err != nil {
+			return fmt.Errorf("error generating event payload: %w", err)
+		}
+		*es = append(*es, Event{Payload: p, URL: webhookURL})
+	}
+	return nil
+}
+
+func getWebhookByName(ctx context.Context, webhookName string, workspaceID int) (*Webhook, error) {
+	webhook := Webhook{}
+	err := db.Bun().NewSelect().
+		Model(&webhook).
+		Relation("Triggers").
+		Where("name = ?", webhookName).
+		Where("workspace_id = ?", workspaceID).
```

Comment:
Nice, this makes sure only one webhook is selected if there are multiple webhooks with the same names but different url.